src/TypesService.ts

```diff
@@ -260,6 +280,15 @@ const emitDtsCommon = (
         async ({ config, project }) => {
           const compilerOptions = project.getCompilerOptions()
           const diagnosticsHost = ts.createCompilerHost(compilerOptions, true)
+          const isIndexReference = (
+            importPath: string,
+            sourceFilename: string | undefined,
+          ): boolean =>
+            sourceFilename !== undefined &&
+            /index\.(js|ts|cjs|mjs|mts|d\.ts|d\.mts)$/.test(
```

Comment:
This reminds me, I need to add JSX support in build-tools

Reply:
I'm pretty sure this includes way more extensions than necessary, but I figured better safe than sorry

Reply:
I like it 👍 there are a few limitations which to be addressed eventually

- Sub-directory entrypoints aren't supported (besides index thanks to this work)
- tsx extension support for relative import/export/module dec remapping

Reply:
> Sub-directory entrypoints aren't supported

Not sure what you mean here. Is it possible to specify something besides `index`? If so I feel like I might as well add it in this PR

Reply:
No, it's not related to this.  In the exports field for multi-export builds it will list every entry point as if it's a src-level entrypoint.  But because the entrypoint is a defined by a glob it could be a nested entrypoint.